internal/crypto/crypto.go

```diff
@@ -157,3 +158,152 @@ func NewEncryptedString(id string, data []byte, keyID string, keyBase64URL strin
 
 	return &es, nil
 }
+
+func VerifySIWS(
```

Comment:
We generally don't do logs like this. Might be a better option to collate all the things that failed and return them as one big error message instead.